src/udp_load_balancer.c

```diff
@@ -21,36 +26,130 @@ unsigned int get_servers_amount(char **servers_list) {
-            servers_hosts[i] = get_ip_from_hostname(servers_hosts[i]);
+    while(1) {
+        servers_amount_resolved_prev = servers_amount_resolved;
+        servers_list_resolved_tmp = calloc(sizeof (char*), servers_amount); 
+        servers_ports_resolved_tmp = calloc(sizeof (int), servers_amount); 
+        if ( verb_flag ) printf("\n\nBalancing to servers:\n");
+        unsigned int i;
+        unsigned int j = 0;
+        for (i = 0; i < servers_amount; ++i) {
+            if ( verb_flag ) printf("  %s\n", servers_list[i]);
+            if (is_ip(servers_hosts[i]) == 0) {
+
+                struct hostent *he = gethostbyname(servers_hosts[i]);
+
+                if (he == NULL) {
+                    printf("Host \"%s\" not found\n", servers_hosts[i]);
+                    continue;
+                }
+
+                struct in_addr **addr_list = (struct in_addr **)he->h_addr_list;
+
+                while ( *addr_list ) {
+                    if ( j >= servers_amount ) {
+                       servers_list_resolved_tmp = realloc(servers_list_resolved_tmp, sizeof (char*) * (j+1));
+                       servers_ports_resolved_tmp = realloc(servers_ports_resolved_tmp, sizeof (int) * (j+1));
+                    }
+                    servers_list_resolved_tmp[j] = strdup(inet_ntoa(**addr_list));
+                    servers_ports_resolved_tmp[j] = servers_ports[i];
+                    if ( verb_flag ) printf("    %s %d\n",servers_list_resolved_tmp[j], servers_ports_resolved_tmp[j]);
+                    j++;
+                    addr_list += 1;
+                }
+            } else {
+              if ( j >= servers_amount ) {
+                 servers_list_resolved_tmp = realloc(servers_list_resolved_tmp, sizeof (char*) * (j+1));
+                 servers_ports_resolved_tmp = realloc(servers_ports_resolved_tmp, sizeof (int) * (j+1));
+              }
+              servers_list_resolved_tmp[j] = strdup(servers_hosts[i]);
+              servers_ports_resolved_tmp[j] = servers_ports[i];
+              j++;
+            }
+            servers_amount_resolved = j;
         }
+        if ( j < 1 ) {
+             printf( "Can't resolve any of servers, waiting for 5sec\n" );
+             sleep(5);
+             continue ;
+        }
+        if ( first_run == 0 ) {
+           pthread_mutex_lock(&resolv_mutex);
+        }
+        first_run = 0;
+        servers_list_resolved_free = servers_list_resolved;
+        servers_list_resolved = servers_list_resolved_tmp;
+        servers_ports_resolved_free = servers_ports_resolved;
+        servers_ports_resolved = servers_ports_resolved_tmp;
+        pthread_mutex_unlock(&resolv_mutex);
+        if ( debug_flag ) {
+          printf("servers_list_resolved array:\n");
+          for ( j = 0 ; j < servers_amount_resolved ; j++ ) {
+               printf(" (%u %s:%d)", j, servers_list_resolved_tmp[j], servers_ports_resolved_tmp[j]);
+               printf("\n");
```

Comment:
why not put it in the same line? (like all the other printf)

Reply:
Yes, perhaps it was left over from cleaning.